### PR TITLE
[IMP] account,*: Peppol First

### PR DESCRIPTION
**Pull request by clbr-odoo**
*: account_edi_ubl_cii, account_peppol

Revamping around Peppol to simplify onboarding.

1. In previous commit [1], we added a footer in email when relevant with some Peppol
related information.
2. Simplify the res.config.setting view.
3. Auto-install the Peppol module for following countries:
AT, BE, CH, CY, CZ, DE, DK, EE, ES, FI, FR, GR, IE, IS, IT, LT,
LU, LV, MT, NL, NO, PL, PT, RO, SE, SI.
4. Peppol Registration wizard: **Remove SMS validation** => we are assessing some
KYC, but the SMS validation is not preventing much, so its only adding frictions.
Remove the migrations key from the wizard, if the user has registered a key through his
previous Access Point, we retrieve it and automatically migrates the participant.
Small UI improvements.
5. Print & Send wizard: refine warnings, disable the "by Peppol" checkbox
if the customer is not on Peppol. Add some modals with explanations about Peppol
if the customer is on Peppol but the Sender didn't check the option.
6. Partner form: Hide in debug some too technical field. Small UI improvements.
7. EAS cleanup: revamp EAS description to be shorter and clearer. Better handling of translations.

[1] https://github.com/odoo/odoo/commit/4e984c0ae234f02bd07d96fc555d5b5e9ad144c3

task-4332306

### CODE REVIEW

**Comment by clbr-odoo** [comment]
TODO update .pot

File: addons/account_edi_ubl_cii/models/res_partner.py [hunk 1]
```diff
@@ -39,81 +39,84 @@ class ResPartner(models.Model):
...
+            ('9946', "9946 - Portugal VAT"),
+            ('9947', "9947 - Romania VAT"),
+            ('9948', "9948 - Serbia VAT"),
+            ('9949', "9949 - Slovenia VAT"),
+            ('9950', "9950 - Slovakia VAT"),
+            ('9952', "9952 - Turkey VAT"),
+            ('9955', "9955 - Swedish VAT"),
+            ('9957', "9957 - French VAT"),
+            ('9959', "9959 - USA EIN"),
+            ('0060', "0060 - DUNS Number"),  # TODO in master, remove all codes below, considered useless/outdated
```
[the remark this answers is not in the document]
Comment (clbr-odoo) [reply addons/account_edi_ubl_cii/models/res_partner.py:103]: As discussed, the ones that are not ordered at the end are meant to be removed in master.

File: addons/account_peppol/models/account_edi_proxy_user.py [hunk 2]
```diff
@@ -282,6 +282,13 @@ def _peppol_get_participant_status(self):
     # BUSINESS ACTIONS
     # -------------------------------------------------------------------------
 
+    @handle_demo
+    def _peppol_migrate_registration(self):
+        self.ensure_one()
+        response = self._call_peppol_proxy(endpoint='/api/peppol/1/migrate_peppol_registration')
```
[the remark this answers is not in the document]
Comment (clbr-odoo) [reply addons/account_peppol/models/account_edi_proxy_user.py:288]: You just retrieve the latest one and override, which I think is fine :thinking:

File: addons/account_peppol/models/account_move_send.py [hunk 3]
```diff
@@ -32,6 +31,30 @@ def _get_alerts(self, moves, moves_data):
                     'message': _("Peppol is in testing/demo mode."),
                     'level': 'info',
                 }
+        if partner_with_peppol_not_selected := moves.filtered(
+            lambda m: m.partner_id.commercial_partner_id.with_company(m.company_id).peppol_verification_state == 'valid'
+                      and 'peppol' not in moves_data[m]['sending_methods']
```
[the remark this answers is not in the document]
Comment (clbr-odoo) [reply addons/account_peppol/models/account_move_send.py:36]: Nice try... :rofl:

[the remark this answers is not in the document]
Comment (smetl) [reply addons/account_peppol/models/account_move_send.py:36]: @clbr-odoo Retry?
```py
def peppol_partner(moves):
    return moves.partner_id.commercial_partner_id

def filter_peppol_state(moves, state):
    return peppol_partner(moves.filtered(lambda m: peppol_partner(m).with_company(m.company_id).peppol_verification_state == state))

# Check for invalid peppol partners.
peppol_moves = moves.filtered(lambda m: 'peppol' in moves_data[m]['sending_methods'])
if (
    invalid_partners := filter_peppol_state(peppol_moves, 'not_valid_format')
    and not 'account_edi_ubl_cii_configure_partner' in alerts
):
    alerts['account_peppol_warning_partner'] = {
        'message': _("Customer is on Peppol but did not enable receiving documents."),
        'action_text': _("View Partner(s)"),
        'action': invalid_partners._get_records_action(name=_("Check Partner(s)")),
    }
    edi_modes = set(
        peppol_moves.company_id.account_edi_proxy_client_ids\
        .filtered(lambda usr: usr.proxy_type == 'peppol')\
        .mapped('edi_mode')
    )
    if edi_modes.intersection({'test', 'demo'}:
        alerts['account_peppol_demo_test_mode'] = {
            'message': _("Peppol is in testing/demo mode."),
            'level': 'info',
        }

# Check for not peppol partners that are on the network.
not_peppol_moves = moves.filtered(lambda m: 'peppol' not in moves_data[m]['sending_methods'])
if peppol_not_selected_partners := filter_peppol_state(not_peppol_moves, 'valid'):
    ...
```

File: addons/snailmail_account/models/account_move_send.py [hunk 4]
```diff
@@ -28,6 +28,7 @@ def _get_alerts(self, moves, moves_data):
     # -------------------------------------------------------------------------
     # HELPERS
     # -------------------------------------------------------------------------
+
```
Comment (smetl) [inline addons/snailmail_account/models/account_move_send.py:31]: useless diff :nerd_face:

**Review by smetl — COMMENTED** [review]
Last comments

File: addons/account_peppol/tests/test_peppol_participant.py [hunk 5]
```diff
@@ -127,6 +125,7 @@ def test_create_success_sender(self):
         # after running the cron checking participant status
         company = self.env.company
         wizard = self.env['peppol.registration'].create(self._get_participant_vals())
+        wizard.smp_registration = False  # by default, we try to register as receiver
```
Comment (smetl) [inline addons/account_peppol/tests/test_peppol_participant.py:128]: You can `create` directly the wizard with the right value for `smp_registration`

File: addons/account_peppol/models/account_move_send.py [hunk 6]
```diff
@@ -13,25 +13,58 @@ class AccountMoveSend(models.AbstractModel):
...
-                    'action': invalid_partners._get_records_action(name=_("Check Partner(s)")),
-                }
-            edi_modes = [move.company_id.account_edi_proxy_client_ids.filtered(lambda usr: usr.proxy_type == 'peppol').edi_mode for move in peppol_moves]
-            if any(edi_mode in ('test', 'demo') for edi_mode in edi_modes):
+        # Check for invalid peppol partners.
+        peppol_moves = moves.filtered(lambda m: 'peppol' in moves_data[m]['sending_methods'])
+        if (
+                (invalid_partners := filter_peppol_state(peppol_moves, 'not_valid_format'))
+                                    and not 'account_edi_ubl_cii_configure_partner' in alerts
+        ):
```
Comment (smetl) [inline addons/account_peppol/models/account_move_send.py:29]: Cosmetic comment: I don't like the `and` in the middle of the condition :nauseated_face:

File: addons/account_peppol/wizard/peppol_registration.py [hunk 7]
```diff
@@ -295,45 +228,66 @@ def button_update_peppol_user_data(self):
...
+                # registration_form_action['views'] = [(False, 'form')]
+                # return {
+                #     'type': 'ir.actions.client',
+                #     'tag': 'display_notification',
+                #     'params': {
+                #         'type': 'danger',
+                #         'message': e,
+                #         'next': registration_form_action,
+                #     }
+                # }
```
Comment (smetl) [inline addons/account_peppol/wizard/peppol_registration.py:280]: Don't forget to clean the commented code before squashing the commits

**Review by smetl — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**